File: JAVA/TAP_Unidad03_Practica03/src/Models/Alumno.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Models;

import java.util.ArrayList;

/**
 *
 * @author Carlos J
 */
public class Alumno {
    
    private String numeroControl;
    private String nombreCompleto;
    private int semestre;
    private Carrera carrera;

    public String getNumeroControl() {
        return numeroControl;
    }

    public void setNumeroControl(String numeroControl) {
        this.numeroControl = numeroControl;
    }

    public String getNombreCompleto() {
        return nombreCompleto;
    }

    public void setNombreCompleto(String nombreCompleto) {
        this.nombreCompleto = nombreCompleto;
    }

    public int getSemestre() {
        return semestre;
    }

    public void setSemestre(int semestre) {
        this.semestre = semestre;
    }

    public Carrera getCarrera() {
        return carrera;
    }

    public void setCarrera(Carrera carrera) {
        this.carrera = carrera;
    }
    
    // Constructor (Snippet ctor + tab)
    public Alumno(){
    }

    // Constructor
    public Alumno(String numControl, String nombreCompleto, int semestre, Carrera carrera){
        this.numeroControl = numControl;
        this.nombreCompleto = nombreCompleto;
        this.semestre = semestre;
        this.carrera = carrera;
    }

    //Busca a un alumno de una lista
    public Alumno buscar(ArrayList<Alumno> lista, String numCont){
        
        for(Alumno calf : lista){
            if (numCont.equalsIgnoreCase(calf.getNumeroControl()))
            {
                return calf;
            }
        }
        
        return null;
    }

    //Decide si se puede eliminar
    public boolean puedeEliminar(String numCont ,ArrayList<Calificaciones> calificaciones){
        for(Calificaciones calf : calificaciones){
            if (numCont.equalsIgnoreCase(calf.getAlumno().getNumeroControl())){
                return false;
            }
        }
        return true;
    }

    //Verifica si un alumno se repite en una lista
    public boolean repite(ArrayList<Alumno> lista, String numCont){
        for(Alumno item : lista){
            if (numCont.equalsIgnoreCase(item.getNumeroControl())){
                return false;
            }
        }
        return true;
    }
    
}

File: C#/TAP_Unidad03_Practica03/TAP_Unidad03_Practica03/Models/Alumno.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace TAP_Unidad03_Practica03.Models
{
    class Alumno
    {
        //Numero de Control
        private String numControl;
        public String NumControl
        {
            get { return numControl; }
            set { numControl = value; }
        }

        //Nombre Completo
        private String nombreCompleto;
        public String NombreCompleto
        {
            get { return nombreCompleto; }
            set { nombreCompleto = value; }
        }

        //Semestre
        private int semestre;
        public int Semestre
        {
            get { return semestre; }
            set { semestre = value; }
        }

        //Carrera
        private Carrera carrera;
        public Carrera Carrera
        {
            get { return carrera; }
            set { carrera = value; }
        }

        // Constructor (Snippet ctor + tab)
        public Alumno()
        {
        }

        // Constructor
        public Alumno(String numControl, String nombreCompleto, int semestre, Carrera carrera)
        {
            this.numControl = numControl;
            this.nombreCompleto = nombreCompleto;
            this.semestre = semestre;
            this.carrera = carrera;
        }

        //Busca a un alumno de una lista
        public Alumno buscar(List<Alumno> lista, String numCont)
        {
            foreach (Alumno encontrado in lista)
            {
                if (numCont == encontrado.numControl)
                {
                    return encontrado;
                }
            }
            return null;
        }

        //Decide si se puede eliminar
        public bool puedeEliminar(String numCont ,List<Calificaciones> calificaciones)
        {
            foreach(Calificaciones calf in calificaciones)
            {
                if (numCont == calf.Alumno.numControl)
                {
                    return false;
                }
            }
            return true;
        }

        public bool repite(List<Alumno> lista, String numCont)
        {
            foreach (Alumno item in lista)
            {
                if (numCont==item.numControl)
                {
                    return false;
                }
            }
            return true;
        }
    }
}
